### ISSUE TITLE
Formatter Issue 

### ISSUE THREAD

**Issue by StanDaMan0505** [Microsoft/vscode-mssql#838]
**Problem Summary**
•MSSQL Extension Version: <0.3.0>
•VSCode Version: <1.11.2>
•OS Version: Windows 10

**Formatter Settings**
 ```
    "mssql.format.keywordCasing": "uppercase",
    "mssql.format.placeSelectStatementReferencesOnNewLine": true
 ```

**Input Text** - place unformatted SQL below
 ```
select top 176
	idh,
	version,
	--case DATEDIFF(minute,STARTTIME, ENDTIME)
	--		when 0 then 1 
	--		else DATEDIFF(minute,STARTTIME, ENDTIME) 
	--end as dauerMin, 
	--case 
	--		when DATEDIFF(SECOND,STARTTIME, ENDTIME) between 0 and 29 then 0 
	--		when DATEDIFF(SECOND,STARTTIME, ENDTIME) between 30 and 59 then 1 
	--		---else concat(convert(nvarchar,DATEDIFF(SECOND,STARTTIME, ENDTIME)/60),':',convert(nvarchar,DATEDIFF(SECOND,STARTTIME, ENDTIME)%60))
	--end as dauerCalc, 
	DATEDIFF(SECOND,STARTTIME, ENDTIME)/60 as [minutos],
	DATEDIFF(SECOND,STARTTIME, ENDTIME)%60 as [secondos],
	STARTTIME,
	ENDTIME,
	SUCCESS,
	NOM
from dbo.mytable
order by 1 desc;
```

**Actual Output** - place formatted SQL (that doesn't work as expected) below
 ```
SELECT top 176 idh,
         version,
         --case DATEDIFF(minute,
        STARTTIME,
         ENDTIME) --
    WHEN 0 THEN
    1 --
    ELSE DATEDIFF(minute,STARTTIME, ENDTIME) --end AS dauerMin, --case --
    WHEN DATEDIFF(SECOND,STARTTIME, ENDTIME)
    BETWEEN 0
        AND 29 THEN
    0 --
    WHEN DATEDIFF(SECOND,STARTTIME, ENDTIME)
    BETWEEN 30
        AND 59 THEN
    1 -- ---else concat(convert(nvarchar,DATEDIFF(SECOND,STARTTIME, ENDTIME)/60),':',convert(nvarchar,DATEDIFF(SECOND,STARTTIME, ENDTIME)%60)) --end AS dauerCalc, DATEDIFF(SECOND,STARTTIME, ENDTIME)/60 AS [minutos], DATEDIFF(SECOND,STARTTIME, ENDTIME)%60 AS [secondos], STARTTIME, ENDTIME, SUCCESS, NOM
FROM dbo.mytable
ORDER BY  1 desc;
```

**Expected Output** - place formatted SQL as you'd expect it to look below
 ```
SELECT TOP 176
	idh,
	version,
	DATEDIFF(SECOND,STARTTIME, ENDTIME)/60 as [minutos],
	DATEDIFF(SECOND,STARTTIME, ENDTIME)%60 as [secondos],
	STARTTIME,
	ENDTIME,
	SUCCESS,
	NOM
FROM dbo.mytable
ORDER BY 1 DESC;
```
And it would be nice if we could define if we like all the columns surrounded with square brackets

**Comment by kevcunnane** [comment]
@StanDaMan0505 thanks for raising this. Regarding square bracket support, that's a great feature suggestion. I've created #839 to track this by itself.

Regarding the main issue: looking at the expected output, the commented code is no longer there. Are you suggesting the formatter should remove it? 
Also, when I paste this into VSCode for myself, I'm not seeing the strange formatting error. Is there more text in the document, or does it happen with this exact input string?

**Comment by StanDaMan0505** [comment]
Hi Kevin,

Thanks for following up on this behavior.

Just right now, I found that the issue was from a different SQL-Prettyfier
I've installed previously.. After that one was uninstalled the behavior is
correct.
The case can be closed.

And thank you for creating an new case for the square brackets. Looking
forward to the future updates.






2017-04-26 22:03 GMT+02:00 Kevin Cunnane <notifications@github.com>:
